CMakeLists.txt

```diff
@@ -270,7 +273,9 @@ else()
   if(WIN32)
     add_compile_flag("-D_GNU_SOURCE")
     add_compile_flag("-D__STDC_FORMAT_MACROS")
-    add_link_flag("-Wl,--stack,8388608")
+    if(NOT CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+      add_link_flag("-Wl,--stack,8388608")
+    endif()
```

Comment:
This change looks unrelated?

Reply:
Ah my bad, should have made a separate PR for this. `--stack` doesn't exist as a linker option on clang, and because clang in mingw64 isn't considered MSVC, this flag is used and causes errors on linking

Reply:
Pushed a new commit to add back the stack, using `-stack_size` which clang supports

Reply:
Can you make this a completely separate PR since it seems unrelated?

Also, I think maybe using `if(MSVC)` is better than checking or `NOT Clang` here?

Reply:
`MSVC` is not set but clang is in my case. Perhaps that's an edge case not worth supporting but I'm using a standard clang install with the msys shell which works for my other libs. Also yeah I'll split that into another PR, how can I do that